Improve `IsEmpty` to check for zero value

danail-branekov opened the pull request:
Instead of checking whether a struct value is the zero time, use reflection to compare it against the actual zero value of its type.
We also noticed that `TestIsEmpty` is expecting `struct{}{}` to be not empty. From our point of view this does not really make sense. Therefore we have amended the test to expect `true` for the empty anonymous struct.

@georgethebeatle

swithek commented:
@danail-branekov do you think it would be possible to increase test coverage here?

danail-branekov commented:
@swithek 
I do not know why the code coverage of this PR is reported as being lower. The PR is mostly deleting code, the code that I have introduced has been covered by tests, both already existing and the ones that I have added.

swithek commented:
@danail-branekov Thanks for the PR. Merging.
